test/Microsoft.AspNet.Mvc.IntegrationTests/BodyValidationIntegrationTests.cs

```diff
@@ -42,11 +42,18 @@ private class Address
                 ParameterType = typeof(Person)
             };
 
-            var operationContext = ModelBindingTestHelper.GetOperationBindingContext();
-            var httpContext = operationContext.HttpContext;
-
-            ConfigureHttpRequest(httpContext.Request, string.Empty);
-            var modelState = new ModelStateDictionary();
+            var operationContext = ModelBindingTestHelper.GetOperationBindingContext(
+              request =>
+              {
+                  request.Body = new MemoryStream(Encoding.UTF8.GetBytes("{ \"Id\":1234 }"));
+                  request.ContentType = "application/json";
+              });
+
+            var actionContext = operationContext
+                .HttpContext
+                .RequestServices
+                .GetRequiredService<IScopedInstance<ActionContext>>().Value;
+            var modelState = actionContext.ModelState;
```

Comment:
Really not a fan of this showing up in tests. What's the need for this? Is this because formatters use the `IScopedInstance<ActionContext>` to access model state? 

This is probably a bug in TryUpdateModel scenarios. See `ModelBindingHelper.TryUpdateModel` - there's an overload which takes a `ModelStateDictionary`